### ISSUE TITLE
chore(android): getDefaultDisplay() and getSize() deprecated in Android SDK 30

### ISSUE THREAD

**Issue by MattGyverLee** [keymanapp/keyman#9586]
The functions getDefaultDisplay() and getSize() have been deprecated for a while,

https://github.com/keymanapp/keyman/blob/31fd6ea42afef1dddd7d883fc31c5086fe9ae9b2/android/Samples/KMSample2/app/src/main/java/com/keyman/kmsample2/SystemKeyboard.java#L181

WindowMetrics seems to be new to 30, and it looks like the deprecation may have happened in 13, so maybe I need an intermediate form to clear the deprecation warning. I'm a bit nervous mucking around in here, but I want to update it in my app. Can someone confirm that this is the code I want in my SystemKeyboard.:

```
if (Build.VERSION.SDK_INT >= 30) {
            WindowMetrics windowMetrics = wm.getCurrentWindowMetrics();
            width = windowMetrics.getBounds().width();
            height = windowMetrics.getBounds().height();
        } else {
            Display display = wm.getDefaultDisplay();
            if(display == null) return;
            display.getSize(size);
            width = size.x;
            height = size.y;
        }
```
LAter, I have: 
```
outInsets.touchableRegion.set(0, outInsets.contentTopInsets, width, height);
```

**Comment by darcywong00** [comment]
Android API 30 Deprecated the following calls:

* [getDefaultDisplay()](https://developer.android.com/reference/android/view/WindowManager#getDefaultDisplay())
* [getSize()](https://developer.android.com/reference/android/view/Display#getSize(android.graphics.Point))
* [getMetrics()](https://developer.android.com/reference/android/view/Display#getMetrics(android.util.DisplayMetrics))  (Used to get density)

**Comment by darcywong00** [comment]
So, this is only available in 18.0.38 alpha  and onward. And we've only just started 18.0 development, so it will be quite a while before KAB gets to this.

> Can someone from Keyman help the Keyboard App Builder devs implement keyboard resizing? 

The Keyman team is also already stretched thin with half of us going on furloughs soon, so unfortunately we won't have availability on this.
